python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+        name="gbdesGlobalAstrometricFit_starCatalog",
+        storageClass="ArrowNumpyDict",
+        dimensions=("instrument", "physical_filter"),
+    )
+
+    def getSpatialBoundsConnections(self):
+        return ("inputVisitSummaries",)
+
+
+class GbdesGlobalAstrometricFitConfig(
+    GbdesAstrometricFitConfig, pipelineConnections=GbdesGlobalAstrometricFitConnections
+):
+    visitOverlap = pexConfig.Field(
+        dtype=float,
+        default=1.0,
+        doc=(
+            "The linkage distance threshold above which clusters of visits "
+            "will not be merged. Calculated using the minimum distance "
+            "between all visits in each set, as a fraction of the "
+            "field-of-view radius"
+        ),
+    )
+
+
+class GbdesGlobalAstrometricFitTask(GbdesAstrometricFitTask):
+    """Calibrate the WCS across multiple visits and multiple fields using the
+    GBDES package.
+    """
+
+    ConfigClass = GbdesGlobalAstrometricFitConfig
+    _DefaultName = "gbdesAstrometricFit"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # We override runQuantum to set up the refObjLoaders
+        inputs = butlerQC.get(inputRefs)
+
+        instrumentName = butlerQC.quantum.dataId["instrument"]
+
+        # Ensure the inputs are in a consistent order
+        inputSumVisits = np.array([inputSum[0]["visit"] for inputSum in inputs["inputVisitSummaries"]])
+        inputs["inputVisitSummaries"] = [inputs["inputVisitSummaries"][v] for v in inputSumVisits.argsort()]
+        inputRefHtm7s = np.array([inputRefCat.dataId["htm7"] for inputRefCat in inputRefs.referenceCatalog])
+        inputRefCatRefs = [inputRefs.referenceCatalog[htm7] for htm7 in inputRefHtm7s.argsort()]
+        inputRefCats = np.array([inputRefCat.dataId["htm7"] for inputRefCat in inputs["referenceCatalog"]])
+        inputs["referenceCatalog"] = [inputs["referenceCatalog"][v] for v in inputRefCats.argsort()]
+        inputISSTracts = np.array(
+            [isolatedStarSource.dataId["tract"] for isolatedStarSource in inputs["isolatedStarSources"]]
+        )
+        inputISCTracts = np.array(
+            [isolatedStarCatalog.dataId["tract"] for isolatedStarCatalog in inputs["isolatedStarCatalogs"]]
+        )
+        for tract in inputISCTracts:
+            if tract not in inputISSTracts:
+                raise RuntimeError(f"tract {tract} in isolated_star_cats but not isolated_star_sources")
+        inputs["isolatedStarSources"] = np.array(
+            [inputs["isolatedStarSources"][t] for t in inputISSTracts.argsort()]
+        )
+        inputs["isolatedStarCatalogs"] = np.array(
+            [inputs["isolatedStarCatalogs"][t] for t in inputISSTracts.argsort()]
+        )
+
+        sampleRefCat = inputs["referenceCatalog"][0].get()
+        refEpoch = sampleRefCat[0]["epoch"]
+
+        refConfig = LoadReferenceObjectsConfig()
+        refConfig.anyFilterMapsToThis = "phot_g_mean"
+        refConfig.requireProperMotion = True
```

Comment:
These are hard-coded because we decided that Gaia is going to be the best astrometric reference for many years to come (and this part of `runQuantum` is copied from the parent task). They could be configurable, but I'm not sure it's worth adding a config that is pretty unlikely to be used.

Reply:
We don't run `gbdesAstrometricFit` on `LSSTCam-imSim` because there are no astrometric effects included in the simulation. This is from the older code, and I'm partly reluctant to change it because I would also want to change the parent class. If it becomes an issue, it would be a quick ticket.

Reply:
Ok, but I would add the operative "currently" to your statement.  Do we know that astrometric effects won't ever be included in sims?  E.g. do we know they won't be included in the simulated ComCam data for the upcoming OPS Commissioning Rehearsal #3: https://confluence.lsstcorp.org/pages/viewpage.action?pageId=239404080 ?

(I'm not going to block you on this, but it does seem like a limitation we may not want to wait to fix before it becomes "urgent"...)